src/openforms/js/components/formio_builder/WebformBuilder.js

```diff
@@ -72,9 +68,94 @@ class WebformBuilder extends WebformBuilderFormio {
       flags
     );
 
-    // Extract the callback that will be called when a component is changed in the editor
     if (!this.editForm) return;
-    const existingOnChangeHandlers = this.editForm.events._events['formio.change'];
+
+    // find the translations containers so that we can populate translations (initially)
+    // and track changes to literals to update the translations. Ideally we'd do this in
+    // the onDrop method, but this one has no extension points and A LOT of code to
+    // copy :(
+    const translationsPathPrefix = 'openForms.translations.';
+    const translationComponents = {};
+    const {componentTranslationsRef} = this.options.openForms;
+
+    // content components have a different layout
+    if (component.type !== 'content') {
+      this.editForm.everyComponent(_component => {
+        if (_component.type !== 'datagrid') return;
+        if (!_component.path.startsWith(translationsPathPrefix)) return;
+
+        const languageCode = _component.path.replace(translationsPathPrefix, '');
+        if (languageCode.includes('.') || languageCode.includes('[')) return; // ensure it's a direct child
+        translationComponents[languageCode] = _component;
+      });
+    }
+
+    // this.editForm is a new instance every time the modal opens, so we can track
+    // some private state on there.
+    if (!this.editForm._ofPreviousLiterals) {
+      this.editForm._ofPreviousLiterals = {};
+    }
+
+    const updateTranslations = translations => {
+      if (component.type === 'content') {
+        Object.entries(translations).forEach(([languageCode, translationsForLanguage]) => {
+          if (!translationsForLanguage.length) return;
+          const {literal, translation} = translationsForLanguage[0];
+
+          // Can't get this to work with FormioUtils.getComponent... it's just skipping
+          // over the columns/hidden fields for some reason :(
+          let literalComponent;
+          let translationComponent;
+          this.editForm.everyComponent(_component => {
+            const prefix = `openForms.translations.${languageCode}[0].`;
```

Comment:
Since this trick of checking the prefix 'openForms.translations.` to make sure we are dealing with the translation component of the edit form is used multiple times, I wonder if it would be worth adding a custom property in the schema of the translation component along the lines of 'isTranslationComponent'?